### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | -

### CODE REVIEW

File: framework/console/CConsoleCommand.php [hunk 1]
```diff
@@ -129,7 +129,13 @@ public function run($args)
 			$name=$param->getName();
 			if(isset($options[$name]))
 			{
-				if($param->isArray())
+				if(version_compare(PHP_VERSION,'8.0','>=')) {
+					$isArray=$param->getType() && $param->getType()->getName()==='array';
+				} else {
+					$isArray = $param->isArray();
+				}
```
Comment (marcovtwout) [inline framework/console/CConsoleCommand.php:136]: Code formatting could be improved a little bit.

File: framework/web/actions/CAction.php [hunk 2]
```diff
@@ -94,7 +94,13 @@ protected function runWithParamsInternal($object, $method, $params)
 			$name=$param->getName();
 			if(isset($params[$name]))
 			{
-				if($param->isArray())
+				if(version_compare(PHP_VERSION,'8.0','>=')) {
+					$isArray=$param->getType() && $param->getType()->getName()==='array';
+				} else {
+					$isArray=$param->isArray();
+                }
```
Comment (marcovtwout) [inline framework/web/actions/CAction.php:101]: Code formatting could be improved a little bit.

File: framework/utils/CPropertyValue.php [hunk 3]
```diff
@@ -53,7 +53,7 @@ class CPropertyValue
 	public static function ensureBoolean($value)
 	{
 		if (is_string($value))
-			return !strcasecmp($value,'true') || $value!=0;
+			return !strcasecmp($value,'true') || ($value!=0 && $value!=='' && is_numeric($value));
```
Comment (marcovtwout) [inline framework/utils/CPropertyValue.php:56]: @samdark Do you have examples of `$value` that are fixed with this change? Just want to make sure this change is correct.

Comment (samdark) [reply framework/utils/CPropertyValue.php:56]: @marcovtwout revert this like in master and run tests on PHP 8. That's covered with tests. It is about https://wiki.php.net/rfc/string_to_number_comparison

**Comment by My6UoT9** [comment]
fyi use `\version_compare` instead of `version_compare`
The later looks up the function in current namespace, and only after finds the correct one.
image

see page 63/64 https://www.slideshare.net/nikita_ppv/php-performance-trivia

**Pull request merged**